app/src/marketplace/components/Modal/ChooseAccessPeriodDialog/index.jsx

```diff
@@ -7,255 +7,222 @@ import cx from 'classnames'
-        getProductPrices()
-    }, [getProductPrices])
+    const displayApproxUsd = useMemo(() => (
+        BN(approxUsd).isNaN() ? 'N/A' : formatDecimals(approxUsd, contractCurrencies.USD)
+    ), [approxUsd])
 
     useEffect(() => {
-        getAccountBalance()
-    }, [getAccountBalance, account])
+        setExternalPrices({
+            dataPerUsd,
+            priceInData,
+            priceInUsd,
+            paymentCurrency,
+        })
+    }, [setExternalPrices, dataPerUsd, priceInData, paymentCurrency, priceInUsd])
 
-    const currentBalance = () => {
-        switch (paymentCurrency) {
-            case paymentCurrencies.DATA:
-                return dataBalance
-            case paymentCurrencies.ETH:
-                return ethBalance
-            case paymentCurrencies.DAI:
-                return daiBalance
-            default:
-                return dataBalance
-        }
-    }
-
-    const price = () => {
-        switch (paymentCurrency) {
-            case paymentCurrencies.DATA:
-                return priceInData
-            case paymentCurrencies.ETH:
-                return priceInEth
-            case paymentCurrencies.DAI:
-                return priceInDai
-            default:
-                return priceInData
-        }
-    }
+    const currentBalance = useMemo(() => (
+        (balances && balances[paymentCurrency]) ? formatDecimals(balances[paymentCurrency], paymentCurrency) : '-'
+    ), [balances, paymentCurrency])
 
     const selectedValue = useMemo(() => options.find(({ value: optionValue }) => optionValue === timeUnit), [timeUnit])
 
     const onTimeUnitChange = useCallback((t) => {
         setTimeUnit(t)
     }, [])
 
-    const onNext = useCallback(async (selectedTime: NumberString | BN, selectedTimeUnit: TimeUnit, selectedPaymentCurrency: PaymentCurrency) => {
-        setLoading(true)
-
-        await onNextProp({
-            time: selectedTime,
-            timeUnit: selectedTimeUnit,
-            paymentCurrency: selectedPaymentCurrency,
-            priceInEth,
-            priceInDai,
-            priceInEthUsdEquivalent: ethPriceInUsd,
-        })
-    }, [ethPriceInUsd, onNextProp, priceInDai, priceInEth])
-
-    const onPaymentCurrencyChange = useCallback((currency: PaymentCurrency) => (setPaymentCurrency(currency)), [setPaymentCurrency])
+    const onPaymentCurrencyChange = useCallback((currency: PaymentCurrency) => {
+        setPaymentCurrency(currency)
+    }, [setPaymentCurrency])
 
     const actions = {
         cancel: {
             title: I18n.t('modal.common.cancel'),
-            onClick: onCancel,
+            onClick: () => onCancel(),
             kind: 'link',
         },
         next: {
             title: I18n.t('modal.common.next'),
             kind: 'primary',
             outline: true,
-            onClick: () => onNext(time, timeUnit, paymentCurrency),
-            disabled: !isValidTime() || !isValidPrice() || loading,
+            onClick: () => onNext({
```

Comment:
removed separate prices from the dialog result, the currency will tell which value it is